nasa-gibs/worldview
# WV-2480: Highlight Event Tracks on Hover

## Description
This feature adds the ability to hover over an event track, an event menu item, or any of the event track points, and display higher visibility for the event track. The line color, line thickness, arrow thickness, and z-index of the track changes when highlighted like this, as well as the menu item in the Event sidebar also appearing highlighted. This option is specifically for use when all event tracks are shown, for visibility and clarity reasons.

## How To Test
1. `git checkout wv-2480-trackhighlight-hover`
2. `npm ci`
3. `npm run watch`
4. Open the Events tab in the sidebar, then click the filter icon to open the Filter Events window. Enable the option for "Show tracks for all events", then click the Apply button to apply the changes.
5. Hover over an event track and verify that the track becomes highlighted (change in line color, line thickness, arrow size, appears above other track lines) and that the menu item in the Event sidebar becomes highlighted. Clicking on the track should also select the event.
6. Hover over the menu item for an event in the Event sidebar and verify that the track becomes highlighted (change in line color, line thickness, arrow size, appears above other track lines) and that the menu item in the Event sidebar becomes highlighted.
7. Hover over an event track point (one of the circles along the event track) and verify that the track becomes highlighted (change in line color, line thickness, arrow size, appears above other track lines) and that the menu item in the Event sidebar becomes highlighted.
8. Hover over an event icon (the icon indicating event type at one end of an event track) and verify that the track becomes highlighted (change in line color, line thickness, arrow size, appears above other track lines) and that the menu item in the Event sidebar becomes highlighted.
